### PR TITLE
Add supported commands in server capabilities

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR updates the server capabilities to include the commands that Ruff supports. This is similar to how there's a list of possible code actions supported by the server.

I noticed this when I was trying to find whether Helix supported workspace commands or not based on Jane's comment (https://github.com/astral-sh/ruff/pull/11831#discussion_r1634984921) and I found the `:lsp-workspace-command` in the editor but it didn't show up anything in the picker.

So, I looked at the implementation in Helix (https://github.com/helix-editor/helix/blob/9c479e6d2de3bca9dec304f9182cee2b1c0ad766/helix-term/src/commands/typed.rs#L1372-L1384) which made me realize that Ruff doesn't provide this in its capabilities. Currently, this does require `ruff` to be first in the list of language servers in the user config but that should be resolved by https://github.com/helix-editor/helix/pull/10176. So, the following config should work:

```toml
[[language]]
name = "python"
# Ruff should come first until https://github.com/helix-editor/helix/pull/10176 is released
language-servers = ["ruff", "pyright"]
```

## Test Plan

1. Neovim's server capabilities output should include the supported commands:

```
  executeCommandProvider = {                                                                                                                          
    commands = { "ruff.applyFormat", "ruff.applyAutofix", "ruff.applyOrganizeImports", "ruff.printDebugInformation" },                                
    workDoneProgress = false                                                                                                                          
  },
```

2. Helix should now display the commands to pick from when `:lsp-workspace-command` is invoked:

Screenshot 2024-06-13 at 08 47 14

### CODE REVIEW

**Comment by dhruvmanila** [comment]
I think I'll re-order them to display the "printDebugInformation" at the end.

File: crates/ruff_server/src/server.rs [hunk 1]
```diff
@@ -354,3 +363,56 @@ impl SupportedCodeAction {
...
     }
 }
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) enum SupportedCommand {
+    Debug,
+    Format,
+    FixAll,
+    OrganizeImports,
+}
```
Comment (dhruvmanila) [inline crates/ruff_server/src/server.rs:373]: I renamed this from `Command` to `SupportedCommand` to match `SupportedCodeAction` and also moved it from the `execute_command.rs` module. Feel free to disagree on anything.

**Review by snowsignal — APPROVED** [review]
LGTM! 

It's unfortunate that Helix isn't able to forward the URI for these commands - maybe that's something we should try contributing upstream eventually.

**Comment by dhruvmanila** [comment]
> It's unfortunate that Helix isn't able to forward the URI for these commands - maybe that's something we should try contributing upstream eventually.

Yeah, we can start by opening an issue / discussion to let them know about this limitation. I also think this can be solved by a custom plugin but I don't think Helix supports them yet.

**Pull request merged**